features/step_definitions/finance/invoicing_steps.rb

```diff
@@ -165,3 +165,9 @@
   nodes = page.find_all(:xpath, ".//tr[contains(@class,'invoice')]/td[contains(text(), '#{date}')]")
   assert_equal 1, nodes.count
 end
+
+Then "the years selector should have the following options:" do |table|
```

Comment:
> There's no such thing as a selector, it's either a select or a dropdown.

I disagree: https://dictionary.cambridge.org/dictionary/english/selector

In this context, selector is just that - some element on the page that allows the user to select something, years in this case. Indeed, it could be implemented as radio buttons, for example.

Although, to be honest, I don't see anything bad in even calling it a "drop-down list". Is there some rule that prohibits referencing UX/UI decisions in cucumber scenarios?

"invoices can be filtered by the following years:" to me it seems a big too vague :shrug:

Reply:
I counter-disagree: https://dictionary.cambridge.org/dictionary/english/drop-down-menu

This is either a dropdown or a select, not a selector. I didn't mean selector does not exist as a word, but as a front-end/web term.

Reply:
It's also worth mentioning that using "should" is discouraged. The steps should be imperative: should vs. is. Moreover, the purpose of this dropdown is to filter invoices by year, therefore I proposed `invoices can be filtered by the following years`.

Reply:
> This is either a dropdown or a select, not a selector. I didn't mean selector does not exist as a word, but as a front-end/web term.

Yeah, OK, but my point is - I used it as a generic word - an element that is used to select some options, not a UI term.

I guess as a team we need to agree whether in our cucumbers we need to be more specific (referencing specific UI elements), or more generic. 

> It's also worth mentioning that using "should" is discouraged. 

When you reference statements like this, can you please also add some references to the documentation or whatever, so we all learn? Who discourages "should"? E.g. the docs here provide quite a lot of examples with "should": https://cucumber.io/docs/gherkin/reference/

Reply:
Of course, sorry. A while ago we talked about "how to write better cucumbers" according to them: https://cucumber.io/docs/bdd/better-gherkin/ and I try to follow these guidelines as much as possible.